tools/compiletest/src/runtest.rs

```diff
@@ -310,6 +324,57 @@ impl<'test> TestCx<'test> {
         self.compose_and_run(kani)
     }
 
+    /// Runs an executable file and:
+    ///  * Checks the expected output if an expected file is specified
+    ///  * Checks the exit code (assumed to be 0 by default)
+    fn run_exec_test(&self) {
+        // Open the `config.yml` file and extract its values
+        let path_yml = self.testpaths.file.join("config.yml");
+        let config_file = std::fs::File::open(path_yml).expect("couldn't open `config.yml`");
+        let exec_config: ExecConfig =
+            serde_yaml::from_reader(config_file).expect("couldn't read `config.yml` values");
```

Comment:
Panicking would make it hard to debug the issues with the yaml file. Perhaps we should emit the error returned by `serde_yaml::from_reader`?

Reply:
Done.